packages/shared/src/hooks/usePersistentContext.ts

```diff
@@ -24,6 +25,14 @@ export default function usePersistentContext<T>(
     getAsyncCache<T>(key, valueWhenCacheEmpty),
   );
 
+  useEffect(() => {
+    const hasDefaultValue = getAsyncCache(key, null);
+    hasDefaultValue.then((defaultValue) => {
+      if (defaultValue !== null) return;
+      queryClient.setQueryData(key, valueWhenCacheEmpty);
+    });
+  }, [valueWhenCacheEmpty]);
+
```

Comment:
A better solution for this would be to use a different query key for `useQuery`. Instead of `key`, we can use `[key, valueWhenCacheEmpty]`. This will make sure to trigger again the function when one of them changes which is exactly what we want to do.